tests/drop_counters/test_drop_counters.py

```diff
@@ -0,0 +1,383 @@
+        src_mac - PTF interface source MAC address
+    """
+    dut_iface = random.choice(tx_dut_ports.keys())
+    ptf_tx_port_id = setup["dut_to_ptf_port_map"][dut_iface]
+    dst_mac = get_dut_iface_mac(duthost, dut_iface)
+    src_mac = ptfadapter.dataplane.ports[(0, ptf_tx_port_id)].mac()
+    return dut_iface, ptf_tx_port_id, dst_mac, src_mac
+
+
+def is_arp_req(pkt, dst_ip):
+    """ Check whether packet is ARP request """
+    arp_request = 1
+    scapy_pkt = scapy.layers.l2.Ether(pkt)
+
+    if scapy_pkt.haslayer(scapy.layers.l2.ARP):
+        if scapy_pkt[scapy.layers.l2.ARP].op == arp_request:
+            if scapy_pkt[scapy.layers.l2.ARP].pdst == dst_ip:
+                return True
+    return False
+
+
+def match_packet_fields(pkt, eth_dst=None, eth_src=None, eth_type=None, ip_dst=None, ip_src=None, tcp_sport=None,
+                        tcp_dport=None):
+    """
+    Verify whether packet fields matches specified fields. Return True for match of all fields, false vise versa
+
+    @param eth_dst - String representation of Ethernet destination MAC address. Template example - "xx:xx:xx:xx:xx:xx"
+    @param eth_src - String representation of Ethernet source MAC address. Template example - "xx:xx:xx:xx:xx:xx"
+    @param eth_type - Hex representation of Ethernet type. Example - '0x800'
+    @param ip_dst - String representation of IP destenation address. Template example - "xxx.xxx.xxx.xxx"
+    @param ip_src - String representation of IP source address. Template example - "xxx.xxx.xxx.xxx"
+    @param tcp_sport - TCP source port (int)
+    @param tcp_dport - TCP destination port (int)
+    """
+    scapy_pkt = scapy.layers.l2.Ether(pkt)
+    match_list = []
+
+    # Check Ethernet layer fields
+    if scapy_pkt.haslayer(scapy.layers.l2.Ether):
+        if eth_dst and scapy_pkt[scapy.layers.l2.Ether].dst != eth_dst:
+            match_list.append(False)
+        if eth_src and scapy_pkt[scapy.layers.l2.Ether].src != eth_src:
+            match_list.append(False)
+        if eth_type and scapy_pkt[scapy.layers.l2.Ether].type != eth_type:
+            match_list.append(False)
+    # Check IP layer fields
+    if scapy_pkt.haslayer(scapy.layers.inet.IP):
+        if ip_dst and scapy_pkt[scapy.layers.inet.IP].dst != ip_dst:
+            match_list.append(False)
+        if ip_src and scapy_pkt[scapy.layers.inet.IP].src != ip_src:
+            match_list.append(False)
+    # Check TCP layer fields
+    if scapy_pkt.haslayer(scapy.layers.inet.TCP):
+        if tcp_sport and scapy_pkt[scapy.layers.inet.TCP].sport != tcp_sport:
+            match_list.append(False)
+        if tcp_dport and scapy_pkt[scapy.layers.inet.TCP].dport != tcp_dport:
+            match_list.append(False)
+
+    if match_list:
+        if all(match_list):
+            return True
+    return False
+
+
+def verify_no_packet_egressed(ptfadapter, eth_dst=None, eth_src=None, ip_dst=None, ip_src=None, eth_type=None, tcp_sport=None,
```

Comment:
Why no to use verify_no_packet from PTF?

Reply:
Updated to use 'testutils.verify_no_packet_any' instead of custom sniffing.